### DISCUSSION TITLE
Output shifting during horizontal scroll

### DISCUSSION THREAD

**Discussion by KaKi87** [sharkdp/bat#2912] [Q&A]
Hello,

When horizontal scroll is enabled with `--wrap=never`, on a file that doesn't need vertical scroll, `bat` starts outputting the file below the currently used height, despite emptying that area, then when horizontally scrolling, shifts the file to the top :

![](https://github.com/sharkdp/bat/assets/21284089/f51d00d8-8202-45e8-8f56-c2af4470bd84)

How to make it directly output the file from the top so that there is no shifting when horizontally scrolling ?

Thanks

**Comment by keith-hall** [comment]
I guess you'd need to ask the developers of the pager you are using - presumably `less`?

**Reply by KaKi87** [reply]
What do you mean ?

**Reply by keith-hall** [reply]
`bat` passes it's output to a program called a pager. It's the pager which is responsible for the weird initial offset/scrolling effect you see. This pager is typically an executable called `less`.

I would be interested to know if you get the same behavior with running `less` on the file directly: `less -S lorem_ipsum.txt` (or perhaps write the `bat` output to a file and run `less` on that, so it includes the "decorations")

**Comment by KaKi87** [comment]
I confirm that `less -S lorem_ipsum.txt` produces the same behavior.

Your wording seem to indicate that the pager can be changed, is that correct, and if yes, how ?

Thanks

**Reply by keith-hall** [reply]
https://github.com/sharkdp/bat?tab=readme-ov-file#using-a-different-pager

**Reply by KaKi87** [reply]
Alright, which value will fix the behavior I reported ?

Thanks

**Reply by keith-hall** [reply]
I unfortunately can't help with that because I have no idea. But I suggest to download and try some modern pagers written in Rust or Golang, see what suits you best 🙂

**Reply by KaKi87** [reply]
Hi,

I [asked on r/LinuxQuestions](https://old.reddit.com/r/linuxquestions/comments/1cr8uto/how_to_prevent_less_s_from_shifting_output_during/) and got an answer : when using `less -Sc lorem_ipsum.txt`, the behavior doesn't occur.

Therefore, I'd like to suggest adding the `c` parameter to `less` when running `bat --wrap=never`.

Thanks

**Reply by KaKi87** [reply]
Hello ?

**Reply by keith-hall** [reply]
I am no pager expert to say if it's a fine solution for all use cases, but feel free to open a PR for discussion:
https://github.com/sharkdp/bat/blob/8f8c953ab69403a7acf516e65265b7f70bc3ec73/src/output.rs#L91-L93

**Reply by KaKi87** [reply]
[Done](https://github.com/sharkdp/bat/pull/2973). Thanks
